### PR TITLE
Migrated Virustotal API from v2 to v3

### PR DESCRIPTION
**Pull request by javiersanchz**
|Related issue|
|---|
| #22573  |

## Description

The VirusTotal API was migrated from `v2` to `v3` following the provided documentation, its functionality was verified [here](https://github.com/wazuh/wazuh/issues/22573#issuecomment-2010296130), and the corresponding tests were launched

## Tests

```console
(unittest-env) wazuh@javier:~/Git/wazuh$ PYTHONPATH=/home/wazuh/Git/wazuh/api:/home/wazuh/Git/wazuh/framework python3 -m pytest integrations/tests/test_virustotal.py 
=========================================================================================== test session starts ============================================================================================
platform linux -- Python 3.10.12, pytest-7.3.1, pluggy-1.4.0
rootdir: /home/wazuh/Git/wazuh/integrations
configfile: pytest.ini
plugins: anyio-4.3.0, aiohttp-1.0.4, trio-0.8.0, html-2.1.1, metadata-3.1.0, asyncio-0.18.1, tavern-1.23.5
asyncio: mode=auto
collected 22 items                                                                                                                                                                                         

integrations/tests/test_virustotal.py ......................                                                                                                                                         [100%]

============================================================================================ 22 passed in 0.10s ============================================================================================
```

All integrations tests:
```console
(unittest-env) wazuh@javier:~/Git/wazuh$ PYTHONPATH=/home/wazuh/Git/wazuh/api:/home/wazuh/Git/wazuh/framework python3 -m pytest integrations/tests/
=========================================================================================== test session starts ============================================================================================
platform linux -- Python 3.10.12, pytest-7.3.1, pluggy-1.4.0
rootdir: /home/wazuh/Git/wazuh/integrations
configfile: pytest.ini
plugins: anyio-4.3.0, aiohttp-1.0.4, trio-0.8.0, html-2.1.1, metadata-3.1.0, asyncio-0.18.1, tavern-1.23.5
asyncio: mode=auto
collected 122 items                                                                                                                                                                                        

integrations/tests/test_maltiverse.py ..............................................................                                                                                                 [ 50%]
integrations/tests/test_pagerduty.py ..........                                                                                                                                                      [ 59%]
integrations/tests/test_shuffle.py ..................                                                                                                                                                [ 73%]
integrations/tests/test_slack.py ..........                                                                                                                                                          [ 81%]
integrations/tests/test_virustotal.py ......................                                                                                                                                         [100%]

=========================================================================================== 122 passed in 2.46s ============================================================================================
```

### CODE REVIEW

**Review by GGP1 — CHANGES_REQUESTED** [review]
Changes in this PR look good to me. 

However, we should update the integration's [documentation](https://documentation.wazuh.com/current/user-manual/capabilities/malware-detection/virus-total-integration.html). There were some changes to the response codes (204 -> 429 and 403 -> 401) that need to be updated in the examples and we should check that everything else is fine too.

File: integrations/tests/test_virustotal.py [hunk 1]
```diff
@@ -266,9 +275,16 @@ def test_request_virustotal_info_md5_after_check_fail_8():
 
 def test_request_virustotal_info_md5_after_check_ok():
     """Test that the md5_after field from alerts are valid md5 hash."""
-    with patch('virustotal.query_api'), patch('virustotal.in_database', return_value=False), patch('virustotal.debug'):
+    with patch('virustotal.query_api', return_value=vt_response_data), \
+         patch('virustotal.in_database', return_value=False), \
```
Comment (GGP1) [inline integrations/tests/test_virustotal.py:279]:
```suggestion
```

File: integrations/tests/test_virustotal.py [hunk 2]
```diff
@@ -98,6 +98,15 @@
...
 
+vt_response_data = {
+        'attributes': {
+            'last_analysis_stats': {
+                'malicious': 2
+            },
+            'sha1': 'valid_sha1_value',
+            'last_analysis_date': 'valid_date_value'
+        }
+    }
```
Comment (GGP1) [inline integrations/tests/test_virustotal.py:109]:
```suggestion
vt_response_data = {
    'attributes': {
        'last_analysis_stats': {
            'malicious': 2
        },
        'sha1': 'valid_sha1_value',
        'last_analysis_date': 'valid_date_value'
    }
}
```

File: integrations/virustotal.py [hunk 3]
```diff
@@ -280,18 +280,19 @@ def query_api(hash: str, apikey: str) -> any:
...
-    headers = {'x-apikey': apikey}
+    headers = {
+        "accept": "application/json",
+        'x-apikey': apikey
+    }
 
     debug('# Querying VirusTotal API')
     response = requests.get(
-        f'https://www.virustotal.com/api/v3/files/{hash}', params=params, headers=headers, timeout=timeout
+        f'https://www.virustotal.com/api/v3/files/{hash}', headers=headers, timeout=timeout
```
Comment (GGP1) [inline integrations/virustotal.py:290]: Remove trailing space

```suggestion
        f'https://www.virustotal.com/api/v3/files/{hash}', headers=headers, timeout=timeout
```

**Review by GGP1 — CHANGES_REQUESTED** [review]

**Review by GGP1 — APPROVED** [review]
LGTM

**Review by Selutario — CHANGES_REQUESTED** [review]
The changes look good so far, but even if the virustotal alerts are being generated, I'm finding this error log in the ossec.log:
```
2024/05/06 11:19:39 wazuh-integratord: ERROR: Unable to run integration for virustotal -> integrations
2024/05/06 11:19:39 wazuh-integratord: ERROR: While running virustotal -> integrations. Output: Exception
 
2024/05/06 11:19:39 wazuh-integratord: ERROR: Exit status was: 1
```

Could you confirm whether this also occurs in your env? 

**Edit:** This is logged in the integrations.log when debug2 is enabled for the integratord service:
```
/tmp/virustotal-1714995152-729810369.alert [REDACTED - VT API KEY]  debug  10 3
# Running VirusTotal script
# Opening alert file at '/tmp/virustotal-1714995152-729810369.alert' with '{'timestamp': '2024-05-06T13:32:31.355+0200', 'rule': {'level': 7, 'description': 'Integrity checksum changed.', 'id': '550', 'mitre': {'id': ['T1565.001'], 'tactic': ['Impact'], 'technique': ['Stored Data Manipulation']}, 'firedtimes': 1, 'mail': False, 'groups': ['ossec', 'syscheck', 'syscheck_entry_modified', 'syscheck_file'], 'pci_dss': ['11.5'], 'gpg13': ['4.11'], 'gdpr': ['II_5.1.f'], 'hipaa': ['164.312.c.1', '164.312.c.2'], 'nist_800_53': ['SI.7'], 'tsc': ['PI1.4', 'PI1.5', 'CC6.1', 'CC6.8', 'CC7.2', 'CC7.3']}, 'agent': {'id': '000', 'name': 'wazuh-master'}, 'manager': {'name': 'wazuh-master'}, 'id': '1714995151.662240', 'cluster': {'name': 'wazuh', 'node': 'master-node'}, 'full_log': "File '/testdir/suspicious-file.exe' modified\nMode: realtime\nChanged attributes: mtime\nOld modification time was: '1714995120', now it is '1714995151'\n", 'syscheck': {'path': '/testdir/suspicious-file.exe', 'mode': 'realtime', 'size_after': '68', 'perm_after': 'rw-r--r--', 'uid_after': '0', 'gid_after': '0', 'md5_after': '44d88612fea8a8f36de82e1278abb02f', 'sha1_after': '3395856ce81f2b7382dee72602f798b642f14140', 'sha256_after': '275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f', 'uname_after': 'root', 'gname_after': 'root', 'mtime_before': '2024-05-06T13:32:00', 'mtime_after': '2024-05-06T13:32:31', 'inode_after': 12647842, 'changed_attributes': ['mtime'], 'event': 'modified'}, 'decoder': {'name': 'syscheck_integrity_changed'}, 'location': 'syscheck'}'
# Requesting VirusTotal information
# Querying VirusTotal API
# Request result from VT server: 1:virustotal:{"virustotal": {"found": 1, "malicious": 1, "source": {"alert_id": "1714995151.662240", "file": "/testdir/suspicious-file.exe", "md5": "44d88612fea8a8f36de82e1278abb02f", "sha1": "3395856ce81f2b7382dee72602f798b642f14140"}, "sha1": "3395856ce81f2b7382dee72602f798b642f14140", "scan_date": 1714995019, "positives": 67, "total": 67, "permalink": "https://www.virustotal.com/gui/file/44d88612fea8a8f36de82e1278abb02f/detection"}, "integration": "virustotal"}
/tmp/virustotal-1714995153-1168759714.alert [REDACTED - VT API KEY]  debug  10 3
# Running VirusTotal script
# Opening alert file at '/tmp/virustotal-1714995153-1168759714.alert' with '{'timestamp': '2024-05-06T13:32:32.897+0200', 'rule': {'level': 12, 'description': 'VirusTotal: Alert - /testdir/suspicious-file.exe - 67 engines detected this file', 'id': '87105', 'mitre': {'id': ['T1203'], 'tactic': ['Execution'], 'technique': ['Exploitation for Client Execution']}, 'firedtimes': 2, 'mail': True, 'groups': ['virustotal'], 'pci_dss': ['10.6.1', '11.4'], 'gdpr': ['IV_35.7.d']}, 'agent': {'id': '000', 'name': 'wazuh-master'}, 'manager': {'name': 'wazuh-master'}, 'id': '1714995152.663031', 'cluster': {'name': 'wazuh', 'node': 'master-node'}, 'decoder': {'name': 'json'}, 'data': {'virustotal': {'found': '1', 'malicious': '1', 'source': {'alert_id': '1714995151.662240', 'file': '/testdir/suspicious-file.exe', 'md5': '44d88612fea8a8f36de82e1278abb02f', 'sha1': '3395856ce81f2b7382dee72602f798b642f14140'}, 'sha1': '3395856ce81f2b7382dee72602f798b642f14140', 'scan_date': '1714995019', 'positives': '67', 'total': '67', 'permalink': 'https://www.virustotal.com/gui/file/44d88612fea8a8f36de82e1278abb02f/detection'}, 'integration': 'virustotal'}, 'location': 'virustotal'}'
# Requesting VirusTotal information
# No syscheck block present in the alert
# Error: Empty message
```

And this in the `ossec.log`:
 ```
2024/05/06 11:35:34 wazuh-integratord[4364] integrator.c:453 at OS_IntegratorD(): DEBUG: # Requesting VirusTotal information

2024/05/06 11:35:34 wazuh-integratord[4364] integrator.c:453 at OS_IntegratorD(): DEBUG: # No syscheck block present in the alert

2024/05/06 11:35:34 wazuh-integratord[4364] integrator.c:453 at OS_IntegratorD(): DEBUG: # Error: Empty message

2024/05/06 11:35:34 wazuh-integratord[4364] integrator.c:453 at OS_IntegratorD(): DEBUG: 

2024/05/06 11:35:34 wazuh-integratord[4364] integrator.c:453 at OS_IntegratorD(): DEBUG: Traceback (most recent call last):

2024/05/06 11:35:34 wazuh-integratord[4364] integrator.c:453 at OS_IntegratorD(): DEBUG:   File "/var/ossec/integrations/virustotal.py", line 362, in <module>

2024/05/06 11:35:34 wazuh-integratord[4364] integrator.c:453 at OS_IntegratorD(): DEBUG:     main(sys.argv)

2024/05/06 11:35:34 wazuh-integratord[4364] integrator.c:453 at OS_IntegratorD(): DEBUG:   File "/var/ossec/integrations/virustotal.py", line 92, in main

2024/05/06 11:35:34 wazuh-integratord[4364] integrator.c:453 at OS_IntegratorD(): DEBUG:     process_args(args)

2024/05/06 11:35:34 wazuh-integratord[4364] integrator.c:453 at OS_IntegratorD(): DEBUG:   File "/var/ossec/integrations/virustotal.py", line 124, in process_args

2024/05/06 11:35:34 wazuh-integratord[4364] integrator.c:453 at OS_IntegratorD(): DEBUG:     raise Exception

2024/05/06 11:35:34 wazuh-integratord[4364] integrator.c:453 at OS_IntegratorD(): DEBUG: Exception

2024/05/06 11:35:34 wazuh-integratord[4364] integrator.c:462 at OS_IntegratorD(): ERROR: Unable to run integration for virustotal -> integrations
2024/05/06 11:35:34 wazuh-integratord[4364] integrator.c:463 at OS_IntegratorD(): ERROR: While running virustotal -> integrations. Output: Exception
 
2024/05/06 11:35:34 wazuh-integratord[4364] integrator.c:464 at OS_IntegratorD(): ERROR: Exit status was: 1
```

Could you also check whether this happens in master branch and 4.8-4.9 too?

**Review by Selutario — APPROVED** [review]
LGTM

**Pull request merged**